query/stdlib/influxdata/influxdb/rules_test.go

```diff
@@ -122,6 +122,216 @@ func TestPushDownRangeRule(t *testing.T) {
 	}
 }
 
+func TestPushDownGroupRule(t *testing.T) {
+	readRange := influxdb.ReadRangePhysSpec{
+		Bucket: "my-bucket",
+		Bounds: flux.Bounds{
+			Start: fluxTime(5),
+			Stop:  fluxTime(10),
+		},
+	}
+
+	tests := []plantest.RuleTestCase{
+		{
+			Name: "simple",
+			// ReadRange -> group => ReadGroup
+			Rules: []plan.Rule{
+				influxdb.PushDownGroupRule{},
+			},
+			Before: &plantest.PlanSpec{
+				Nodes: []plan.Node{
+					plan.CreateLogicalNode("ReadRange", &readRange),
+					plan.CreateLogicalNode("group", &universe.GroupProcedureSpec{
+						GroupMode: flux.GroupModeBy,
+						GroupKeys: []string{"_measurement", "tag0", "tag1"},
+					}),
+				},
+				Edges: [][2]int{{0, 1}},
+			},
+			After: &plantest.PlanSpec{
+				Nodes: []plan.Node{
+					plan.CreatePhysicalNode("ReadGroup", &influxdb.ReadGroupPhysSpec{
+						ReadRangePhysSpec: readRange,
+						GroupMode:         flux.GroupModeBy,
+						GroupKeys:         []string{"_measurement", "tag0", "tag1"},
+					}),
+				},
+			},
+		},
+		{
+			Name: "with successor",
+			// ReadRange -> group -> count  =>  ReadGroup -> count
+			Rules: []plan.Rule{
+				influxdb.PushDownGroupRule{},
+			},
+			Before: &plantest.PlanSpec{
+				Nodes: []plan.Node{
+					plan.CreateLogicalNode("ReadRange", &readRange),
+					plan.CreateLogicalNode("group", &universe.GroupProcedureSpec{
+						GroupMode: flux.GroupModeBy,
+						GroupKeys: []string{"_measurement", "tag0", "tag1"},
+					}),
+					plan.CreatePhysicalNode("count", &universe.CountProcedureSpec{}),
+				},
+				Edges: [][2]int{
+					{0, 1},
+					{1, 2},
+				},
+			},
+			After: &plantest.PlanSpec{
+				Nodes: []plan.Node{
+					plan.CreatePhysicalNode("ReadGroup", &influxdb.ReadGroupPhysSpec{
+						ReadRangePhysSpec: readRange,
+						GroupMode:         flux.GroupModeBy,
+						GroupKeys:         []string{"_measurement", "tag0", "tag1"},
+					}),
+					plan.CreatePhysicalNode("count", &universe.CountProcedureSpec{}),
+				},
+				Edges: [][2]int{{0, 1}},
+			},
+		},
+		{
+			Name: "with multiple successors",
+			//
+			// group    count       group    count
+			//     \    /       =>      \    /
+			//    ReadRange            ReadRange
+			//
+			Rules: []plan.Rule{
+				influxdb.PushDownGroupRule{},
+			},
+			Before: &plantest.PlanSpec{
+				Nodes: []plan.Node{
+					plan.CreateLogicalNode("ReadRange", &readRange),
+					plan.CreateLogicalNode("group", &universe.GroupProcedureSpec{
+						GroupMode: flux.GroupModeBy,
+						GroupKeys: []string{"_measurement", "tag0", "tag1"},
+					}),
+					plan.CreatePhysicalNode("count", &universe.CountProcedureSpec{}),
+				},
+				Edges: [][2]int{
+					{0, 1},
+					{0, 2},
+				},
+			},
+			After: &plantest.PlanSpec{
```

Comment:
You can use the `NoChange: true` field to indicate that a rule should not be applied, I find it makes tests a bit easier to read.